ddtrace/contrib/aws_lambda/patch.py

```diff
@@ -20,14 +20,17 @@ def _crash_flush(_, __):
     Tags the current root span with an Impending Timeout error.
     Finishes spans with ancestors from the current span.
     """
-
     root_span = tracer.current_root_span()
-    root_span.error = 1
-    root_span.set_tag_str(ERROR_MSG, "Datadog detected an Impending Timeout")
-    root_span.set_tag_str(ERROR_TYPE, "Impending Timeout")
+    if root_span is not None:
+        root_span.error = 1
+        root_span.set_tag_str(ERROR_MSG, "Datadog detected an Impending Timeout")
+        root_span.set_tag_str(ERROR_TYPE, "Impending Timeout")
+    else:
+        log.warning("An impending timeout was reached, but no root span was found. No error will be tagged.")
```

Comment:
Is this unexpected?  When aws lambda is used should there always be an active root span? 
```suggestion
```

Reply:
Yes, this is unexpected, since the `datadog-lambda` package always creates a wrapper span titled `aws.lambda`.

Reply:
Update: customer explained how their lambda behaved and it doesn't appear to be falling in the description above, we might have to follow up asking them to file a support ticket to understand why this might be happening to them. Nonetheless, this is log is useful on its own.